### PR TITLE
Add FSS abstraction to remaining `from_*` IO functions

### PR DESCRIPTION
**Pull request by philss**
This is a continuation of #645 

PS: I left out the introduction of `endpoint / base_url` to another PR.

### CODE REVIEW

File: lib/explorer/data_frame.ex [hunk 1]
```diff
@@ -682,34 +690,40 @@ defmodule Explorer.DataFrame do
...
+    {:ok, %S3.Entry{url: entry, config: config}}
   end
 
-  defp normalise_entry("file://" <> path, _config), do: %Local.Entry{path: path}
+  defp normalise_entry("file://" <> path, _config), do: {:ok, %Local.Entry{path: path}}
 
   defp normalise_entry(filepath, _config) when is_binary(filepath) do
     if File.exists?(filepath) do
-      %Local.Entry{path: filepath}
+      {:ok, %Local.Entry{path: filepath}}
```
Comment (josevalim) [inline lib/explorer/data_frame.ex:721]: What if we always return it as `%Local.Entry{path: filepath}`? Then if it doesn't exist, Polars will fail to read anyway and say the file does not exist. This way we don't need to fiddle with ArgumentError below. Thoughts?

**Review by josevalim — APPROVED** [review]
One tiny comment and ship it!

File: lib/explorer/data_frame.ex (hunk 1 above)
Comment (philss) [reply lib/explorer/data_frame.ex:721]: But you mean, remove the result from the `normalise_entry/2` entirely? Or just always return `{:ok, %Local.Entry{path: filepath}}` for this function clause?

Comment (josevalim) [reply lib/explorer/data_frame.ex:721]: Don't check if the file exists, alwaus return `{:ok, ...}` for the function clause.

**Pull request merged**